arwen/contexts/runtime.go

```diff
@@ -33,10 +33,7 @@ type runtimeContext struct {
 }
 
 // NewRuntimeContext creates a new runtimeContext
-func NewRuntimeContext(
-	host arwen.VMHost,
-	vmType []byte,
-) (*runtimeContext, error) {
+func NewRuntimeContext(host arwen.VMHost, vmType []byte, protocolReservedFunctions []string) (*runtimeContext, error) {
```

Comment:
I suggest not passing `protocolReservedFunctions` as argument directly to this constructor, but instead the constructor should retrieve these function names from the `host`, with a getter. See `host.GetAPIMethods()` and its use in `NewRuntimeContext()` as an example.

Reply:
Fixed.